### DISCUSSION TITLE
How to safely update server via docker?

### DISCUSSION THREAD

**Discussion by krisnoble** [music-assistant/support#1438] [Q&A] [answered]
Hi folks, wondering if I can get some guidance on how to properly update the server via a docker container please? Apologies if this is a noob-level question!

I have the following bash script which I amended from the one I use for HA, but on running it today although it worked I lost all my settings, library etc, which doesn't happen in HA. As far as I can tell from reading around the `pull -> stop -> rm -> run` sequence is the standard way to do this, did I do something wrong?

I was updating from 2.0.0b62 to 2.0.0b67 in case that's relevant.

```
docker pull ghcr.io/music-assistant/server:latest
docker stop mass
docker rm mass
docker run -d --name mass --network host --privileged -v /home/pi/Music:/Music ghcr.io/music-assistant/server

**Comment by OzGav** [comment, marked as answer]
I think there was answer on discord about this. It was said

`make sure the datadir is a peristent volume if you're using docker yourself. If you look for more easy of maintenance, maybe Home Assistant OS is a good candidate as an all-in-one appliance.`

**Reply by krisnoble** [reply]
Thank you, that seems to have done the trick. HA OS is not an option at this time but adding `-v <dir>:/data` to my `docker run` command has given me a persistent data store. I think this would be useful to add to the docs so will put in a PR. Thanks again!

**Comment by oneofthemany** [comment]
as a worthwhile consideration, Podman  allows you to use systemd to manage an auto update service.

This way you dont have to worry about doing anything, just create a file after running this command editing where you /data volume exists outside the container:

`export volume=/opt/music_assistant/data # insert your volume for data persistence`

```
cat <<EOF >> /etc/systemd/system/container-music-assistant.service
# container-music-assistant.service
[Unit]
Description=Podman container-music-assistant.service
Documentation=man:podman-generate-systemd(1)
Wants=network.target
After=network-online.target

[Service]
Environment=PODMAN_SYSTEMD_UNIT=%n
Restart=on-failure
TimeoutStopSec=70
ExecStartPre=/bin/rm -f %t/container-music-assistant.pid %t/container-music-assistant.ctr-id
ExecStart=/usr/bin/podman run --conmon-pidfile %t/container-music-assistant.pid --cidfile %t/container-music-assistant.ctr-id --cgroups=no-conmon --replace -d --name music-assistant --network host --privileged -v ${volume}:/data --label "io.containers.autoupdate=image" ghcr.io/music-assistant/server
ExecStop=/usr/bin/podman stop --ignore --cidfile %t/container-music-assistant.ctr-id -t 10
ExecStopPost=/usr/bin/podman rm --ignore -f --cidfile %t/container-music-assistant.ctr-id
PIDFile=%t/container-music-assistant.pid
Type=forking

[Install]
WantedBy=multi-user.target default.target
EOF
```
then

```
systemctl daemon-reload
systemctl enable container-music-assistant
systemctl start container-music-assistant
```

**Reply by krisnoble** [reply]
Interesting, thank you! Was not aware of Podman so will have to look into it more. I feel like I still prefer having full control over exactly when it updates and to what version for things like HA and MA, but looks very useful for other projects 👍🏻
